sources/packages/api/src/services/application/application.service.ts

```diff
@@ -176,7 +197,23 @@ export class ApplicationService extends RecordDataModelService<Application> {
       application.studentAssessments = [originalAssessment];
       application.currentAssessment = originalAssessment;
 
-      await this.repo.save(application);
+      // When application and assessment is saved, assess for SIN restriction.
```

Comment:
Small language adjustment: "When application and assessment **are** saved, assess for SIN restriction."